agent/runner/jobs/pbm_job_logger.go

```diff
@@ -0,0 +1,113 @@
+// Copyright 2022 Percona LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package jobs
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"net/url"
+	"sync/atomic"
+	"time"
+
+	"google.golang.org/protobuf/types/known/timestamppb"
+
+	"github.com/percona/pmm/api/agentpb"
+)
+
+type pbmJob string
+
+const (
+	pbmBackupJob  pbmJob = "backup"
+	pbmRestoreJob pbmJob = "restore"
+)
+
+type pbmJobLogger struct {
+	dbURL      *url.URL
+	jobID      string
+	logChunkID uint32
+}
+
+func newPbmJobLogger(jobID string, mongoURL *url.URL) *pbmJobLogger {
+	return &pbmJobLogger{
+		jobID:      jobID,
+		logChunkID: 0,
+		dbURL:      mongoURL,
+	}
+}
+
+func (l *pbmJobLogger) sendLog(send Send, data string, done bool) {
+	send(&agentpb.JobProgress{
+		JobId:     l.jobID,
+		Timestamp: timestamppb.Now(),
+		Result: &agentpb.JobProgress_Logs_{
+			Logs: &agentpb.JobProgress_Logs{
+				ChunkId: atomic.AddUint32(&l.logChunkID, 1) - 1,
+				Data:    data,
+				Done:    done,
+			},
+		},
+	})
+}
+
+func (l *pbmJobLogger) streamLogs(ctx context.Context, send Send, jobType pbmJob, name string) error {
```

Comment:
Same with `send` and `jobType`

Reply:
Got it, I've moved `jobType` to constructor. `send` is currently passed to the `Run` method only, and not initialized when we call constructor though. We can refactor it but it leads to too many changes and I don't think it's worth it at the moment.

Reply:
Ok-doki. Let's think about it later.